torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -18,6 +18,19 @@ c10::intrusive_ptr<ProcessGroup::Work> broadcast(
           root_rank, root_tensor, std::chrono::milliseconds(timeout)});
 }
 
+c10::intrusive_ptr<ProcessGroup::Work> allreduce_(
+    at::TensorList tensors,
+    const c10::intrusive_ptr<ProcessGroup>& process_group,
+    int64_t reduce_op,
+    int64_t timeout) {
+  auto tensor_vec = tensors.vec();
+  return process_group->allreduce(
+      tensor_vec,
+      AllreduceOptions{
+          static_cast<ReduceOp>(reduce_op),
```

Comment:
any reason here for using a static cast instead of directly passing in the reduce op as an argument?

Reply:
ReduceOp is an enum class and reduce_op is a int64_t. I think a cast is needed here.